### PR TITLE
fix: access restrictions for brig timers

### LINKED ISSUE #5314: Prisoners in jail cells (and just about anyone else) are able to adjust the time on cell timers to let themselves out.
## Description
The time on cell timers can be reduced using the right click context menu. This is good. However, anyone is able to do this, and prisoners are often able to reach cell timers from inside their cells. This means that prisoners can lower their own sentence and let theirselves out. The ability for anyone to use brig timers could be solved by making it so that adjusting timers requires security access.

### Steps to Reproduce
1.  Set timer on cell timer. Go into relevant jail cell.
2.  From inside jail cell, right click on the timer and reduce your own sentence.
3.  Watch as door opens in response to timer counting down to 0. You're free now.

### PR DESCRIPTION
**Pull request by JustinTrotter**
### Purpose
The time on cell timers can be reduced using the right click context menu. This is good. However, anyone is able to do this, and prisoners are often able to reach cell timers from inside their cells. This means that prisoners can lower their own sentence and let themselves out. The ability for anyone to use brig timers could be solved by making it so that adjusting timers requires security access.

Fixes #5314 

### Changelog:
CL: fix: access restrictions for brig timers

### CODE REVIEW

**Comment by Bod9001** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- UnityProject/Assets/Scripts/Objects/StatusDisplay.cs  2
         

Clones removed
==============
+ UnityProject/Assets/Scripts/Objects/StatusDisplay.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6403559&bid=20757495)

File: UnityProject/Assets/Scripts/Objects/StatusDisplay.cs [hunk 1]
```diff
@@ -58,6 +58,19 @@ public enum MountedMonitorState
 		private bool multiMaster = true;
 		public bool MultiMaster => multiMaster;
 
+		private AccessRestrictions accessRestrictions;
+		public AccessRestrictions AccessRestrictions
+		{
+			get
+			{
+				if (!accessRestrictions)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Objects/StatusDisplay.cs]: explictly compare to null instead ``accessRestrictions == null`` . This is mostly so new contributors understand what is happening here.

File: UnityProject/Assets/Scripts/Objects/StatusDisplay.cs [hunk 2]
```diff
@@ -228,20 +241,28 @@ public void ServerPerformInteraction(HandApply interaction)
 				{
 					if (channel == StatusDisplayChannel.DoorTimer)
 					{
-						currentTimerSeconds += 60;
-						if (currentTimerSeconds > 600)
+						if (!AccessRestrictions || AccessRestrictions.CheckAccess(interaction.Performer))
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Objects/StatusDisplay.cs]: same here, compare to null

File: UnityProject/Assets/Scripts/Objects/StatusDisplay.cs [hunk 3]
```diff
@@ -228,20 +241,28 @@ public void ServerPerformInteraction(HandApply interaction)
...
-						if (!countingDown)
-						{
-							StartCoroutine(TickTimer());
+							currentTimerSeconds += 60;
+							if (currentTimerSeconds > 600)
+							{
+								currentTimerSeconds = 1;
+							}
+
+							if (!countingDown)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Objects/StatusDisplay.cs]: Explicitly compare to false here. I personally dislike it, but it is part of our code style guidelines.

File: UnityProject/Assets/Scripts/Objects/StatusDisplay.cs [hunk 4]
```diff
@@ -228,20 +241,28 @@ public void ServerPerformInteraction(HandApply interaction)
...
+							}
+							else
+							{
+								OnTextBroadcastReceived(StatusDisplayChannel.DoorTimer);
+							}
 						}
 						else
 						{
-							OnTextBroadcastReceived(StatusDisplayChannel.DoorTimer);
+							Chat.AddExamineMsg(interaction.Performer, $"Access Denied.");
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Objects/StatusDisplay.cs:263]: This is a just a suggestion, could maybe also a play a sound here.

Comment (JustinTrotter) [reply UnityProject/Assets/Scripts/Objects/StatusDisplay.cs:263]: I'll see about finding the most annoying sound. Bad enough to give reason for officers to increase the time.

**Comment by Bod9001** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- UnityProject/Assets/Scripts/Objects/StatusDisplay.cs  2
         

Clones removed
==============
+ UnityProject/Assets/Scripts/Objects/StatusDisplay.cs  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6403559&bid=20757495)

**Pull request merged**